app/models/job.ts

```diff
@@ -408,6 +416,62 @@ export class Job extends Record {
     return terminalStates.includes(this.status);
   }
 
+  /**
+   * Checks whether sharing of this job is restricted by any EULAs for
+   * any collection used by this job.
+   * Defaults to true if any collection does not have the harmony.has-eula tag
+   * associated with it.
+   * @param accessToken - the token to make the request with
+   * @returns true or false
+   */
+  async collectionsHaveEulaRestriction(accessToken: string): Promise<boolean> {
+    const cmrCollections = await getCollectionsByIds(this.collectionIds, accessToken, 'harmony.has-eula');
+    if (cmrCollections.length !== this.collectionIds.length) {
+      return true;
+    }
+    return !cmrCollections.every((collection) => (collection.tags
+      && collection.tags['harmony.has-eula'].data === false));
```

Comment:
I see the tests for collection has no `harmony.has-eula` tag set so it looks like this line does default to not allowing access unless the tag is explicitly set:

```
collection.tags['harmony.has-eula'].data === false
```